nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/PutAzureCosmosDBRecord.java

```diff
@@ -189,18 +281,33 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 }
                 batch.add(contentMap);
                 if (batch.size() == ceiling) {
-                    bulkInsert(batch);
+                    bulkInsert(batch, partitionKeyField);
                     batch = new ArrayList<>();
                 }
             }
             if (!error && batch.size() > 0) {
-                bulkInsert(batch);
+                bulkInsert(batch, partitionKeyField);
+            }
+        } catch (CosmosException e)  {
+            final int statusCode =  e.getStatusCode();
+            logger.error("statusCode: " + statusCode + ", subStatusCode: "  + e.getSubStatusCode());
+
+            if (statusCode == 429) {
+                logger.error("Failure due to server-side throttling. Increase RU setting for your workload");
+                yield = true;
+            } else if (statusCode == 410) {
+                logger.error("A request to change the throughput is currently in progress");
+                yield = true;
+            } else {
+                error = true;
             }
-        } catch (SchemaNotFoundException | MalformedRecordException | IOException | CosmosException e) {
+        } catch (IllegalTypeConversionException | ProcessException | SchemaNotFoundException | MalformedRecordException | IOException e) {
             logger.error("PutAzureCosmoDBRecord failed with error: {}", new Object[]{e.getMessage()}, e);
```

Comment:
I think some of the `logger.error()` above might be duplicated here since the error messages are coming from the exception. Do we need both?